pkg/remotewrite/config.go

```diff
@@ -110,6 +124,22 @@ func (conf Config) RemoteConfig() (*remote.HTTPConfig, error) {
 		hc.TLSConfig.Certificates = []tls.Certificate{cert}
 	}
 
+	if isSigV4PartiallyConfigured(conf.SigV4Region, conf.SigV4AccessKey, conf.SigV4SecretKey) {
+		return nil, errors.New(
+			"sigv4 seems to be partially configured. All of " +
+				"K6_PROMETHEUS_RW_SIGV4_REGION, K6_PROMETHEUS_RW_SIGV4_ACCESS_KEY, K6_PROMETHEUS_RW_SIGV4_SECRET_KEY " +
+				"must all be set. Unset all to bypass sigv4",
+		)
+	}
+
+	if conf.SigV4Region.Valid && conf.SigV4AccessKey.Valid && conf.SigV4SecretKey.Valid {
+		hc.SigV4 = &sigv4.Config{
+			Region:             conf.SigV4Region.String,
+			AwsAccessKeyID:     conf.SigV4AccessKey.String,
+			AwsSecretAccessKey: conf.SigV4SecretKey.String,
+		}
+	}
```

Comment:
`isSigV4PartiallyConfigured` seems like a functional copy of `sigv4.Config.validate()`, it has a minor UX benefit that it explicitly refers to environment variables, but still, to be honest, it's better to have a single version so I have a minor suggestion to get rid of it and just make `sigv4.Config.validate()` exported and execute here, something like:

```suggestion
	if conf.SigV4Region.Valid && conf.SigV4AccessKey.Valid && conf.SigV4SecretKey.Valid {
		hc.SigV4 = &sigv4.Config{
			Region:             conf.SigV4Region.String,
			AwsAccessKeyID:     conf.SigV4AccessKey.String,
			AwsSecretAccessKey: conf.SigV4SecretKey.String,
		}
		
		if err := hc.SigV4.Validate(); err != nil {
		    return nil, fmt.Errorf(
			"sigv4 seems to be partially configured. All of " +
				"K6_PROMETHEUS_RW_SIGV4_REGION, K6_PROMETHEUS_RW_SIGV4_ACCESS_KEY, K6_PROMETHEUS_RW_SIGV4_SECRET_KEY " +
				"must all be set: %w.", err),
		}
	}
```

How does that sound?

Reply:
The observation I have is that user can set `K6_PROMETHEUS_RW_SIGV4_ACCESS_KEY` and `K6_PROMETHEUS_RW_SIGV4_ACCESS_KEY` but forgot to set the region env variable. Now they will get 403 error but the error will be coming from upstream and won't be caught early since this condition will resolve to false ` conf.SigV4Region.Valid && conf.SigV4AccessKey.Valid && conf.SigV4SecretKey.Valid`

Reply:
Good point! :+1: